src/controller/python/chip-device-ctrl.py

```diff
@@ -558,6 +550,7 @@ def wait_for_one_discovered_device(self):
         while (not self.devCtrl.GetIPForDiscoveredDevice(0, addrStrStorage, strlen) and count < maxWaitTime):
             time.sleep(0.2)
             count = count + 0.2
+        return count != maxWaitTime
```

Comment:
```suggestion
        return count < maxWaitTime
```
With floating-point numbers you never know if A + A == 2A :)